Add Zenoh router integration (#2254)
DataDog/integrations-extras: zenoh_router/README.md

# Agent Check: Zenoh router

## Overview

This check monitors Zenoh router.

[Zenoh][1] is open source Zero Overhead Network Protocol.

Zenoh /zeno/ is a pub/sub/query protocol unifying data in motion, data at rest and computations. It elegantly blends traditional pub/sub with geo distributed storage, queries and computations, while retaining a level of time and space efficiency that is well beyond any of the mainstream stacks.

Zenoh router integration allows get statistic from zenoh router and monitor router/peer/client connection statuses.

## Setup

### Installation from the datadog

> For Agent v7.21+ / v6.21+, follow the instructions below to install Zenoh router check on your host.

Reviewer: I recommend omitting this line, since the heading and the edits below now cover it.

> 1. Run the following command to install the Agent integration:

Reviewer: ```suggestion
1. On your host, run the following command to install the Agent integration:
```

   ```shell
   datadog-agent integration install -t datadog-zenoh_router==<INTEGRATION_VERSION>
   ```

2. Configure your integration similar to core [integrations][11].

### Installation from the source code

To install the Zenoh router check on your host:

1. Install the [developer toolkit][12] on any machine.

2. Run `ddev release build zenoh_router` to build the package.

> 3. [Download the Datadog Agent][3].

Reviewer: I recommend omitting this step in favor of the edits to step 4 below, since the download destination is confusing here.

> 4. Upload the build artifact to any host with an Agent and

Reviewer: ```suggestion
3. Upload the build artifact to any host with [the Agent installed][3] and
```

>  run `datadog-agent integration install -w

Reviewer: I recommend breaking this command out into its own step, and starting it with "On the host," to make it clear where to run the command.

 path/to/zenoh_router/dist/<ARTIFACT_NAME>.whl`.

### Configuration

1. Make sure that [Zenoh REST API plugin][2] is enabled 

2. Edit the `zenoh_router.d/conf.yaml` file in the `conf.d/` folder at the root of your [Agent's configuration directory][10] to start collecting your Zenoh router [metrics](#metrics).
See the [sample zenoh_router.d/conf.yaml][4] for all available configuration options.

> 3. [Restart the Agent][5]

Reviewer: ```suggestion
3. [Restart the Agent][5].
```
